torch/fx/experimental/symbolic_shapes.py

```diff
@@ -282,14 +297,18 @@ def _maybe_evaluate_static(self, expr: "sympy.Expr") -> "Optional[sympy.Expr]":
         """
         Tries to evaluate expr without introducing guards
         """
+        expr = self.simplify(expr)
         # Simplifies assuming that shape vars > 1 (since we cache on 0/1 shape values)
         symbols = list(expr.free_symbols)
         new_shape_env = {
             k: sympy.Symbol(f"shape_{idx}", positive=True, integer=True) + 1
             for idx, k in enumerate(symbols)
         }
         new_expr = expr.xreplace(new_shape_env)
-        new_expr = sympy.expand(new_expr)
+        floor_div_replace = {}
+        for atom in new_expr.atoms(FloorDiv):
+            floor_div_replace[atom] = atom.args[0] // atom.args[1]
+        new_expr = sympy.expand(new_expr.xreplace(floor_div_replace))
```

Comment:
Why is this sound? args are sympy expressions right? How come we didn't use // in the first place then?

Reply:
Because `a // b` in sympy maps to `floor(a / b)`, which maps to `floor(mul(a, pow(b, -1)))`, and that's too annoying to substitute lol.